mops/include/mops/capi.hpp

```diff
@@ -25,4 +25,17 @@ MOPS_EXPORT const std::string &get_last_error_message();
         }                                                                      \
     } while (false)
 
+#define MOPS_CATCH_EXCEPTIONS_WITH_RETURN(__code__)                            \
+    do {                                                                       \
+        try {                                                                  \
+            return __code__;                                                   \
+        } catch (const std::exception &e) {                                    \
+            mops::store_error_message(e.what());                               \
+            return nullptr;                                                    \
+        } catch (...) {                                                        \
+            mops::store_error_message("unknown error");                        \
+            return nullptr;                                                    \
+        }                                                                      \
+    } while (false)
+
```

Comment:
I think we changed this since you started working on the CUDA branch. You can have a look at main, where we split this into a `CATCH_EXCEPTIONS_BEGIN` and `CATCH_EXCEPTIONS_END`, mainly because it results in more readable code when formatted with clang_format. I think it should be possible to do the same here (?), and I hope without too much effort. I'm also pinging @Luthaf